### PR TITLE
[Compose] - 2606 - Add the mute option to ChannelInfo dialog

### PR DESCRIPTION
**Pull request by bychkovdmitry**
https://github.com/GetStream/stream-chat-android/issues/2606

### 🎯 Goal

Add the "Mute"/"Unmute" channel option to the channel actions dialog.

### 🎨 UI Changes

https://user-images.githubusercontent.com/9600921/141699165-8ed6651e-ae88-4978-9100-df1b18b6aebd.mov

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [x] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS (task created))
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-compose/src/main/res/values-ko/strings.xml [hunk 1]
```diff
@@ -15,8 +15,10 @@
     <!-- Channel Info Dialog -->
     <string name="stream_compose_channel_info_view_info">정보 보기</string>
     <string name="stream_compose_channel_info_leave_group">그룹 나가기</string>
-    <string name="stream_compose_channel_info_leave_group_confirmation_title">Leave group</string>
+    <string name="stream_compose_channel_info_leave_group_confirmation_title">그룹을 떠나다</string>
     <string name="stream_compose_channel_info_leave_group_confirmation_message">%1$s 그룹에서 나가시겠습니까?</string>
+    <string name="stream_compose_channel_info_mute_channel">음소거 채널</string>
```
Comment (skydoves) [inline stream-chat-android-compose/src/main/res/values-ko/strings.xml:20]: 채널 음소거

File: stream-chat-android-compose/src/main/res/values-ko/strings.xml [hunk 2]
```diff
@@ -15,8 +15,10 @@
     <!-- Channel Info Dialog -->
     <string name="stream_compose_channel_info_view_info">정보 보기</string>
     <string name="stream_compose_channel_info_leave_group">그룹 나가기</string>
-    <string name="stream_compose_channel_info_leave_group_confirmation_title">Leave group</string>
+    <string name="stream_compose_channel_info_leave_group_confirmation_title">그룹을 떠나다</string>
```
Comment (skydoves) [inline stream-chat-android-compose/src/main/res/values-ko/strings.xml:18]: 그룹 나가기

**Review by skydoves — APPROVED** [review]
Looks good to me!
I left some comments for Korean translations.

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channel/ChannelListViewModel.kt [hunk 3]
```diff
@@ -298,9 +320,15 @@ public class ChannelListViewModel(
      */
     private fun enrichMutedChannels(channels: List<Channel>, mutedChannelIds: List<String>): List<Channel> {
         val mutedChannelIdsSet = mutedChannelIds.toSet()
-        channels.forEach { channel ->
-            channel.isMuted = channel.id in mutedChannelIdsSet
+        return channels.map { channel ->
+            val isMuted = channel.id in mutedChannelIdsSet
+
+            if (channel.isMuted != isMuted) {
+                channel.copy(extraData = channel.extraData.toMutableMap())
```
Comment (filbabic) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channel/ChannelListViewModel.kt:327]: Why do we have to transform the extraData here?

Comment (bychkovdmitry) [reply stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channel/ChannelListViewModel.kt:327]: 1 out of 10 times doesn't work without extra data transformation. Compose just doesn't see the change:
1. Lets imagine we have a channel in the list
2. When mutes are updated we make a shallow copy of the channel with `channel.copy`
3. Both channels reference the very same extra data
4. Updating "muted" field in extra data updates it for both channels
5. As a result, recomposition is not triggered

Comment (filbabic) [reply stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channel/ChannelListViewModel.kt:327]: Thanks, that makes sense!

**Review by filbabic — APPROVED** [review]
LGTM!

**Review by skydoves — APPROVED** [review]
💯

**Pull request merged**